receiver/scraperhelper/settings.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package scraperhelper // import "go.opentelemetry.io/collector/receiver/scraperhelper"
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/multierr"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+var (
+	ErrNonPositiveInterval    = errors.New("requires positive value")
+	ErrTimeoutExceedsInterval = errors.New("timeout value exceeds collection interval")
+)
+
+// ScraperControllerSettings defines common settings for a scraper controller
+// configuration. Scraper controller receivers can embed this struct, instead
+// of receiver.Settings, and extend it with more fields if needed.
+type ScraperControllerSettings struct {
```

Comment:
Separate issue to rename this to maybe `ScraperControllerConfig`? or just `Config`.

Reply:
It is fine as is, it is wordy, but I don't believe there is value in changing the name since it is already adopted within Contrib.

Reply:
The value is for consistency with other places, since this is a very experimental package. Anyway if you don't want to do it it is ok, will do later.